From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sat, 4 Apr 2020 08:54:33 +0200
Subject: [PATCH] HACK: make neon support optional

It fails to build with older ARMv5 compilers and is useless anyways on CPUs
that don't support it

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 nss/lib/freebl/Makefile | 3 +++
 nss/lib/freebl/gcm.c    | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/nss/lib/freebl/Makefile b/nss/lib/freebl/Makefile
index ce9d36f3a37c..b8200d5b66c2 100644
--- a/nss/lib/freebl/Makefile
+++ b/nss/lib/freebl/Makefile
@@ -124,6 +124,8 @@ ifeq ($(CPU_ARCH),aarch64)
     EXTRA_SRCS += aes-armv8.c gcm-aarch64.c
 endif
 ifeq ($(CPU_ARCH),arm)
+ifdef USE_NEON
+    DEFINES += -DUSE_NEON
     EXTRA_SRCS += gcm-arm32-neon.c
     ifdef CC_IS_CLANG
         DEFINES += -DUSE_HW_AES
@@ -140,6 +142,7 @@ ifeq ($(CPU_ARCH),arm)
         endif
     endif
 endif
+endif
 
 ifeq ($(OS_TARGET),OSF1)
     DEFINES += -DMP_ASSEMBLY_MULTIPLY -DMP_NO_MP_WORD
diff --git a/nss/lib/freebl/gcm.c b/nss/lib/freebl/gcm.c
index 2a42f74c0e3a..66eaa38ce9e7 100644
--- a/nss/lib/freebl/gcm.c
+++ b/nss/lib/freebl/gcm.c
@@ -21,7 +21,7 @@
 #if defined(__aarch64__) && defined(IS_LITTLE_ENDIAN) && \
     (defined(__clang__) || defined(__GNUC__) && __GNUC__ > 6)
 #define USE_ARM_GCM
-#elif defined(__arm__) && defined(IS_LITTLE_ENDIAN)
+#elif defined(__arm__) && defined(IS_LITTLE_ENDIAN) && defined(USE_NEON)
 /* We don't test on big endian platform, so disable this on big endian. */
 #define USE_ARM_GCM
 #endif
